src/main/scala/EmrSparkPlugin.scala

```diff
@@ -307,7 +307,14 @@ object EmrSparkPlugin extends AutoPlugin {
     val jar = assembly.value
     val s3Jar = new S3Url(sparkS3JarFolder.value) / jar.getName
     log.info(s"Putting ${jar.getPath} to ${s3Jar.toString}")
-    sparkS3ClientBuilder.value.build().putObject(s3Jar.bucket, s3Jar.key, jar)
+    if (!sparkS3SSEEnabled.value) {
+      sparkS3ClientBuilder.value.build().putObject(s3Jar.bucket, s3Jar.key, jar)
+    } else {
+      val objectMetadata = new ObjectMetadata
+      objectMetadata.setSSEAlgorithm(ObjectMetadata.AES_256_SERVER_SIDE_ENCRYPTION)
+      val putRequest = new PutObjectRequest(s3Jar.bucket, s3Jar.key, jar)
+      sparkS3ClientBuilder.value.build().putObject(putRequest)
```

Comment:
Should this be
```scala
sparkS3ClientBuilder.value.build().putObject(putRequest.withMetadata(objectMetadata))
```
?

Reply:
You are correct somehow I missed that . Good Catch!. Will make the change and commit. Thnx